tools/dm/skip-replace-sqls.md

```diff
@@ -0,0 +1,547 @@
+
+**Notes:**
+
+- You can only register one operator (specified by `--binlog-pos`) for one binlog event. The previous one can be overwritten by the newly registered operator.
+- Do not specify an operator for one binlog event by using `--binlog-pos` and `--sql-pattern` at the same time.
+- After the operator successfully matches with the binlog event (not after the execution succeeds), it will be deleted. If you need to match again (using `--sql-pattern`) later, you have to register a new operator.
+
+### Support Scenarios
+
+- During the replication, if the DDL statement unsupported by TiDB is executed in the upstream and replicated to the downstream, the replication task will get interrupted.
+    - If it is acceptable that this DDL statement is skipped in the downstream TiDB, then you can use `sql-skip` to resume the replication.
+    - If it is acceptable that this DDL statement is replaced with other DDL statements, then you can use `sql-replace` to resume the replication.
+- During the replication, if you know in advance that an unsupported SQL statement is going to be replicated, you can handle it beforehand to avoid breaking the replication.
+    - If it is acceptable that this DDL statement is skipped in the downstream TiDB, then you can use `sql-skip` to preset an operation to automatically skip this DDL statement when it needs to be executed.
+    - If it is acceptable that this DDL statement is replaced with other DDL statements, then you can use `sql-replace` to preset an operation to automatically replace this DDL statement when it needs to be executed.
+
+### Implementation Principles
+
+In DM, simplified procedures of incremental data replication can be described as follows:
+
+1. The relay processing unit is used as a slave of the upstream MySQL to fetch the binlog that is persisted in the local storage as the relay log. 
+2. The binlog replication unit (sync) reads the local relay log to obtain the binlog event. 
+3. The binlog replication unit parses the binlog event and builds the DDL/DML statements, and then replicates these statements to the downstream TiDB.
+
+When the binlog replication unit is parsing the binlog event and replicating data to the downstream, the replication process might get interrupted because the corresponding SQL statement is not supported by TiDB.
+
+In DM, you can register some skip/replace operators for the binlog event. Before replicating the SQL statements to the downstream, you can compare the current binlog event information(position, DDL statement) with registered operators. If the position or the DDL matches with a registered operator, you can execute the operation corresponding to the operator and then remove this operator. 
```

Comment:
```suggestion
In DM, you can register some skip/replace operators for the binlog event. Before replicating the SQL statements to the downstream, DM will compare the current binlog event information(position, DDL statement) with registered operators. If the position or the DDL matches with a registered operator, it will execute the operation corresponding to the operator and then remove this operator. 
```